sdk/servicebus/service-bus/src/serializers/queueResourceSerializer.ts

```diff
@@ -128,87 +130,73 @@ export interface QueueOptions {
    */
   lockDuration?: string;
 
-  /**
-   * The entity's size in bytes.
-   *
-   */
-  sizeInBytes?: number;
-
   /**
    * Specifies the maximum queue size in megabytes. Any attempt to enqueue a message that
    * will cause the queue to exceed this value will fail.
    */
   maxSizeInMegabytes?: number;
 
   /**
-   * The entity's message count.
+   * If enabled, the topic will detect duplicate messages within the time
+   * span specified by the DuplicateDetectionHistoryTimeWindow property.
+   * Settable only at queue creation time.
    */
-  messageCount?: number;
+  requiresDuplicateDetection?: boolean;
```

Comment:
I understand the need to maintain expected order when building the http request and so `buildQueueOptions()` ensures that is done.

But, what is the need to re-order the members of the `QueueOptions` interface here?
This adds noise to the diff view and makes it harder than it needs to be to review.

Please consider not re-ordering the members of any of the interfaces.

Reply:
There's a 1:1 mapping between user input options and the internal options. I think it makes it cleaner and easier to maintain for all future updates to them i.e., helps avoid misses to updates. 
This was a problem earlier and harder to find as we have intriduced different naming for the same properties (defaultMessageTtl for instance)

Reply:
In the future consider having a separate commit with the refactoring/rearranging of the code to make the review simpler .